app/assets/javascripts/type_form.js

```diff
@@ -0,0 +1,13 @@
+jQuery(document).ready(function($) {
+  jQuery("[id^=type_attribute_visibility_default]").change(function() {
+    var active = jQuery(this);
+    var alwaysId = active.attr("id").replace(/_default_/, "_visible_");
+    var always = jQuery("#" + alwaysId);
+
+    if (this.checked) {
+      always.attr("disabled", false);
+    } else {
+      always.attr("disabled", true);
+    }
```

Comment:
Please use `.prop(attr, boolean)`(http://api.jquery.com/prop/#prop1)

Also, this can be reduced to 

`always.prop('disabled', !this.checked);`